src/algebra/pointwise.lean

```diff
@@ -755,50 +751,53 @@ protected def has_mul [has_mul α] : has_mul (finset α) :=
 
 localized "attribute [instance] finset.has_mul finset.has_add" in pointwise
 
+section has_mul
+variables [has_mul α] {s s₁ s₂ t t₁ t₂ : finset α}
```

Comment:
I would lean towards leaving `s s₁ s₂ t t₁ t₂` lemma-local, as it makes it easier to spot explicit vs implicit, and reorder arguments. It also prevents `variables` lines accumulating new variables that end up unused.

Reply:
I'm arguing this is a matter of taste, as for me not seeing a variable declared in a statement means it's implicit. And the accumulation of variables that end up unused is easy to avoid (just check from time to time), innocuous, and in my experience pretty rare anyway.